parser/ini_test.go

```diff
@@ -0,0 +1,436 @@
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got, _ := parser.GetSection("Deployment")
+		want := map[string]string{
+			"cpu":       "4",
+			"memory":    "8192",
+			"name":      "peertest",
+			"project":   "peertube",
+			"public_ip": "true",
+		}
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+	t.Run("getsection_fromfile", func(t *testing.T) {
+		parser := NewParser()
+		parser.ReadFromFile(file_path)
+		got, _ := parser.GetSection("Profile")
+		want := map[string]string{
+			"name":     "jarvis",
+			"password": "secret",
+		}
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+}
+
+func TestGetSections(t *testing.T) {
+	t.Run("getsections_fromstring", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got := parser.GetSections()
+		want := []string{"Profile", "Deployment", "Owner"}
+
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+	t.Run("getsections_fromfile", func(t *testing.T) {
+		parser := NewParser()
+		parser.ReadFromFile(file_path)
+		got := parser.GetSections()
+		want := []string{"Profile", "Deployment", "Owner"}
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+}
+func TestGetSectionKeys(t *testing.T) {
+	t.Run("getsectionkeys_fromstring", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got := parser.GetSectionKeys("Profile")
+		want := []string{"name", "password"}
+
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+	t.Run("getsection_fromfile", func(t *testing.T) {
+		parser := NewParser()
+		parser.ReadFromFile(file_path)
+		got := parser.GetSectionKeys("Owner")
+		want := []string{"name", "email"}
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+}
+
+func TestWriteToFile(t *testing.T) {
+	t.Run("writetofile_fromstring", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got := parser.WriteToFile("./testdata/writetofile_string_sample.ini")
```

Comment:
use a go tempfile functionality to write to a temp path instead of hardcoding it here, because this can pollute the repo.

Reply:
using the tempfile functionality will create new files every test run, unlike what's happening now it overrides the existing file